### PR TITLE
Add system test to test the AWS auth manager

### PR DESCRIPTION
**Pull request by vincbeck**
Add system test to test the AWS auth manager. These tests are not run as part of the Airflow CI. They are using real AWS credentials to test the AWS auth manager.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/www/security_manager.py
Comment (vincbeck) [inline airflow/www/security_manager.py]: These methods were removed in #37889 but they are actually needed

File: airflow/providers/amazon/aws/auth_manager/views/auth.py
Comment (o-nikolas) [inline airflow/providers/amazon/aws/auth_manager/views/auth.py:96]: Is this a bugfix being included with the new tests?

File: airflow/www/security_manager.py
Comment (o-nikolas) [reply airflow/www/security_manager.py]: Should these be added back in a separate PR?

File: tests/system/providers/amazon/aws/tests/test_aws_auth_manager.py
Comment (o-nikolas) [inline tests/system/providers/amazon/aws/tests/test_aws_auth_manager.py:74]: Should we have a dependency on the production file? I don't have the context to say one way or another. Just calling it out.

File: tests/system/providers/amazon/aws/tests/test_aws_auth_manager.py
Comment (o-nikolas) [inline tests/system/providers/amazon/aws/tests/test_aws_auth_manager.py]: debug statement?

File: tests/system/providers/amazon/aws/tests/test_aws_auth_manager.py
Comment (o-nikolas) [inline tests/system/providers/amazon/aws/tests/test_aws_auth_manager.py:210]: Why should this be in the response after the login? Is this detecting that you land on the airflow home page with the DAGs listed?

File: airflow/providers/amazon/aws/auth_manager/views/auth.py
Comment (vincbeck) [reply airflow/providers/amazon/aws/auth_manager/views/auth.py:96]: Correct

File: tests/system/providers/amazon/aws/tests/test_aws_auth_manager.py
Comment (vincbeck) [reply tests/system/providers/amazon/aws/tests/test_aws_auth_manager.py]: Oopsie! Good catch!

File: tests/system/providers/amazon/aws/tests/test_aws_auth_manager.py
Comment (vincbeck) [reply tests/system/providers/amazon/aws/tests/test_aws_auth_manager.py:74]: Yes, this is not a big deal. It is in the same provider package, and the usage makes sense

File: tests/system/providers/amazon/aws/tests/test_aws_auth_manager.py
Comment (vincbeck) [reply tests/system/providers/amazon/aws/tests/test_aws_auth_manager.py:210]: Correct :)

File: airflow/www/security_manager.py
Comment (vincbeck) [reply airflow/www/security_manager.py]: I can do that if you prefer, I dont have strong opinion on that one. But yeah, that would make sense. I can do that

File: airflow/providers/amazon/aws/auth_manager/views/auth.py
Comment (o-nikolas) [reply airflow/providers/amazon/aws/auth_manager/views/auth.py:96]: Maybe also another PR? Wont push too hard though, I suppose it's fine if they're low impact.

Comment (vincbeck) [reply airflow/providers/amazon/aws/auth_manager/views/auth.py:96]: Yeah I supposed it made sense to include the fix as part of the same PR as the test? But I can also create a separate PR for it. I am fine either way

File: airflow/www/security_manager.py
Comment (vincbeck) [reply airflow/www/security_manager.py]: #37997

Comment (vincbeck) [reply airflow/www/security_manager.py]: I converted this PR as a draft, waiting for #37997 to be merged first. I'll rebase this PR and put it back in review mode

Comment (vincbeck) [reply airflow/www/security_manager.py]: Done

**Review by o-nikolas — APPROVED** [review]

**Pull request merged**
